driver.go

```diff
@@ -632,35 +618,85 @@ func memoryInBytes(strmem string) (int64, error) {
 	}
 }
 
-func parseImage(image string) (string, error) {
-	// strip http/https and docker transport prefix
-	for _, prefix := range []string{"http://", "https://", "docker://"} {
-		if strings.HasPrefix(image, prefix) {
-			image = strings.Replace(image, prefix, "", 1)
+// Creates the requested image if missing from storage
+// returns the 64-byte image ID as an unique image identifier
+func (d *Driver) createImage(image string) (string, error) {
+	var imageID string
+	imageName := image
+	// If it is a shortname, we should not have to worry
+	// Let podman deal with it according to user configuration
+	if !shortnames.IsShortName(image) {
+		imageRef, err := parseImage(image)
+		if err != nil {
+			return imageID, fmt.Errorf("invalid image reference %s: %w\n", image, err)
 		}
+		// If its non-docker transport, we cannot ask for a pull or
+		// check for existence in the API without image plumbing
+		if imageRef.Transport().Name() != "docker" {
+			if imageRef.Transport().Name() != "oci-archive" &&
+				imageRef.Transport().Name() != "docker-archive" {
+				return imageID, fmt.Errorf("unssupported transport: %s", imageRef.Transport().Name())
+			}
```

Comment:
This was a bit hard for me to parse, do you mind condensing it into a switch to handle the error instead?
```suggestion
		switch name := imageRef.Transport().Name(); name {
		case "docker", "oci-archive", "docker-archive":
		default:
			return imageID, fmt.Errorf("unssupported transport: %s", imageRef.Transport().Name())
		}

```

Reply:
Thank you for this suggestion! It looks so much nicer as a switch.

I've changed it a bit to remove the default case. We shouldn't deal with unsupported transports here but in `parseImage`.